### ISSUE TITLE
I want to implement an IDEA plug-in based on HA to achieve remote hot deployment

### ISSUE THREAD

**Issue by KevinClair** [HotswapProjects/HotswapAgent#618]
Hello, HA is a very useful tool for me. When using IDEA Plugin, I found that it can only realize hot deployment of local language.， such as [HotSwapHelper](https://github.com/gejun123456/HotSwapHelper)Now I want to develop an IDEA Plugin based on HA to realize remote hot deployment. For example, select a few files in IDEA Plugin and push them to the remote Linux server (maybe a specific directory), and then HA will update the class file after monitoring the file changes. Is there any solution to realize this function?

**Comment by skybber** [comment]
Great idea! I don’t know of any such solution, nor do I know how to implement deployment to a remote environment (using HA?). Otherwise, part of the problem is addressed by a debugger that can update classes in a remote JVM, but it doesn’t save them to .class files and can’t add new classes. So, such a plugin would definitely be interesting and would expand the current capabilities of remote debugging.

**Comment by KevinClair** [comment]
When starting a java application on a Linux server, and starting HA via -javaagent, is it possible to update .class files by placing the changed files in a special path and then have HA monitor changes to the files in this path?

**Comment by skybber** [comment]
Yes, it is possible. There are 2 configuration parameters - `extraClasspath` and `autoHotswap` designed for this purpose. Look at :

https://hotswapagent.org/mydoc_configuration.html
